public/app/features/alerting/unified/components/expressions/Expression.tsx

```diff
@@ -57,6 +58,16 @@ export const Expression: FC<ExpressionProps> = ({
 
   const queryType = query?.type;
 
+  const { setError, clearErrors } = useFormContext();
+
+  const onQueriesValidationError = useCallback(
+    (errorMsg: string | undefined) => {
+      errorMsg && setError('queries', { type: 'custom', message: errorMsg });
+      !errorMsg && clearErrors('queries');
```

Comment:
if we setErrors on the form manually, we need to clear them manually

Reply:
The reason behind this change is that for Hysteresis expression, we are not adding an actual field in the form but we want to validate it as it was a separate field. 
The only way I found to achieve this,  was calling setError from the FromContext, and as a consequence of this manual setError , we need to clear the error manually.

For more information about how `SetError` works see [documentation](https://react-hook-form.com/docs/useform/seterror)

In rules section you will see:

> An error that is not associated with an input field will be persisted until cleared with clearErrors. This behaviour is only applicable for built-in validation at field level.